sdk/src/Services/SQS/Custom/Model/GetQueueAttributesResponse.extensions.cs

```diff
@@ -181,32 +181,31 @@ public bool FifoQueue
             {
                 var attributeValue = getAttributeValue(SQSConstants.ATTRIBUTE_FIFO_QUEUE);
 
-                if (attributeValue == null || attributeValue.Trim() == string.Empty)
-                {
-                    return false;
-                }
-
                 bool result;
-                return bool.TryParse(attributeValue, out result);
+                if (bool.TryParse(attributeValue, out result))
+                    return result;
+                return false;
             }
         }
 
         /// <summary>
-        /// Whether or not the First-In-First-Out (FIFO) queue has content based deduplication enabled
+        /// Whether or not the First-In-First-Out (FIFO) queue has content based deduplication enabled. For non-FIFO queues this will return null.
         /// </summary>
-        public bool ContentBasedDeduplication
+        public bool? ContentBasedDeduplication
         {
             get
             {
                 var attributeValue = getAttributeValue(SQSConstants.ATTRIBUTE_CONTENT_BASED_DEDUPLICATION);
 
                 if (attributeValue == null || attributeValue.Trim() == string.Empty)
                 {
-                    return false;
+                    return null;
                 }
 
                 bool result;
-                return bool.TryParse(attributeValue, out result);
+                if (bool.TryParse(attributeValue, out result))
+                    return result;
+                return false;
```

Comment:
If we were not able to parse the service response, we should be returning `null`.